src/components/picker/helpers/usePickerMode.ts

```diff
@@ -0,0 +1,47 @@
+import {ExpandableOverlayProps} from 'src/incubator';
+import {PickerProps, PickerModeTypes} from '../index';
+import {CustomPickerProps} from '../types';
+
+type PickerType = {
+  dialog?: boolean;
+  wheelPicker?: boolean;
+  modal?: boolean;
+  custom?: boolean;
+};
+
+const usePickerMode = (props: PickerProps) => {
+  let type: PickerType = {dialog: false, wheelPicker: false, modal: false, custom: false};
+  let headerProps: any;
+  let renderCustomModal: CustomPickerProps['renderCustomModal'];
+  let dialogProps: ExpandableOverlayProps['dialogProps'];
+  let pickerModalProps: ExpandableOverlayProps['modalProps'];
+  if ('pickerType' in props) {
+    const {pickerType} = props;
+    if (pickerType) {
+      type[pickerType] = true;
+    }
+    switch (pickerType) {
+      case PickerModeTypes.Modal:
+        headerProps = props.headerProps;
+        //@ts-ignore
+        pickerModalProps = props.pickerModalProps || props.modalProps;
+        break;
+      case PickerModeTypes.Dialog:
+      case PickerModeTypes.WheelPicker:
+        headerProps = props.headerProps;
+        dialogProps = headerProps && {...props?.customPickerProps?.dialogProps, headerProps: props.headerProps};
+        break;
+      case PickerModeTypes.Custom:
+        renderCustomModal = props.renderCustomModal;
+        break;
+    }
+    return {type, headerProps, renderCustomModal, dialogProps, pickerModalProps};
+  } else {
+    const {useDialog, useWheelPicker} = props;
+    type = {dialog: !!useDialog, wheelPicker: !!useWheelPicker, modal: !useDialog && !useWheelPicker, custom: false};
```

Comment:
If there's no 'pickerType' you're not using the renderCustomModal, dialogProps or pickerModalProps?

Reply:
Yes you are right.
I expect that `headerProps` and `renderCustomModal` will be passed for a specific `pickerType` so if the user doesn't passing `pickerType` I'm not using those props.

Reply:
Why custom is always false? If the user passed renderCustomModal can't we make the type be 'custom' and not default to 'modal'?

Reply:
fixed 
```
custom: !!renderCustomModal
```